src/server.rs

```diff
@@ -950,6 +949,22 @@ impl<B: IntoBuf> SendResponse<B> {
             .map_err(Into::into)
     }
 
+
+    /// Push a request and response to the client
+    ///
+    /// On success, a [`SendResponse`] instance is returned.
+    ///
+    /// [`SendResponse`]: #
+    pub fn push_request(
+        &mut self,
+        request: Request<()>,
+    ) -> Result<SendResponse<B>, ::Error> {
```

Comment:
Since this returns a `SendResponse`, it looks like we will have a similar problem as with the client half.

Only "root" streams may have push promises. Calling `push_request` on the returned `SendResponse` should fail. Is there a test for this?

Or should this function return `SendPushedResponse`, a type similar to `SendResponse` but without a `push_request` function?

Reply:
My plan was definitely the `SendPushedResponse`! I think that's the cleanest option.